### PR TITLE
Fix Sonar collector refresh logic

### PR DESCRIPTION
**Pull request by benj58xu**
Sonar collector refresh logic did not work for collector items just created, before any scheduled collection activities.

### CODE REVIEW

**Review by nameisaravind — APPROVED** [review]
lgtm

File: src/main/java/com/capitalone/dashboard/controller/SonarController.java [hunk 1]
```diff
@@ -75,35 +71,30 @@ public SonarController(SonarProjectRepository sonarProjectRepository,
...
+                        findSonarProjectsByProjectName(collectorId, instanceUrl, projectName);
+                if (CollectionUtils.isNotEmpty(sonarProjects)) {
+                    sonarProjects = sonarProjects.stream().filter(p -> p.getProjectId()==null || project.getProjectId().equals(p.getProjectId())).collect(Collectors.toList());
+                }
+                if (CollectionUtils.isNotEmpty(sonarProjects)) {
+                    Collections.sort(sonarProjects, Comparator.comparing(SonarProject::getLastUpdated).reversed());
+                    SonarProject existingProject = sonarProjects.get(0);
+                    existingProject.setProjectId(project.getProjectId());
+                    return existingProject;
+                }
```
Comment (Sbrenthughes) [inline src/main/java/com/capitalone/dashboard/controller/SonarController.java:95]: Any reason for the duplicate if cases?

Comment (benj58xu) [reply src/main/java/com/capitalone/dashboard/controller/SonarController.java:95]: Not a duplicate. The variable sonarProjects get further filtered (reassigned).

**Review by Sbrenthughes — APPROVED** [review]

File: src/main/java/com/capitalone/dashboard/controller/SonarController.java (hunk 1 above)
Comment (Sbrenthughes) [reply src/main/java/com/capitalone/dashboard/controller/SonarController.java:95]: Ahhh makes sense. Approved.

**Pull request merged**